### PR TITLE
Add toggle password to Password section in settings.

### PR DESCRIPTION
**Pull request by Anupam-dagar**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
This PR adds toggle password visibility in Password Section of Settings.

#### Changes proposed in this pull request:

- Add a toggle password visibility button

Fixes: #2477

## Screenshots
Screenshot from 2019-04-05 21-18-54
Screenshot from 2019-04-05 21-19-02

### CODE REVIEW

File: app/templates/components/settings/password-section.hbs [hunk 1]
```diff
@@ -3,21 +3,40 @@
     <label>
       {{t 'Current Password'}}
     </label>
-    {{input type='password' name='password_current' placeholder=(t 'Current Password') value=passwordCurrent}}
+    {{#if showPass}}
```
Comment (CosmicCoder96) [inline app/templates/components/settings/password-section.hbs]: use inline if.

File: app/templates/components/settings/password-section.hbs [hunk 2]
```diff
@@ -3,21 +3,40 @@
...
+    {{/if}}
   </div>
   <div class="field">
     <label>
       {{t 'New Password'}}
     </label>
-    {{input type='password' name='password_new' placeholder=(t 'New Password') value=passwordNew}}
+    {{#if showPass}}
+      {{input type='text' name='password_new' placeholder=(t 'New Password') value=passwordNew}}
+    {{else}}
```
Comment (CosmicCoder96) [inline app/templates/components/settings/password-section.hbs]: use inline if.

File: app/templates/components/settings/password-section.hbs [hunk 3]
```diff
@@ -3,21 +3,40 @@
...
+      {{input type='password' name='password_new' placeholder=(t 'New Password') value=passwordNew}}
+    {{/if}}
   </div>
   <div class="field">
     <label>
       {{t 'Repeat Password'}}
     </label>
-    {{input type='password' name='password_repeat' placeholder=(t 'Repeat Password') value=passwordRepeat}}
+    {{#if showPass}}
+      {{input type='text' name='password_repeat' placeholder=(t 'Repeat Password') value=passwordRepeat}}
```
Comment (CosmicCoder96) [inline app/templates/components/settings/password-section.hbs]: use inline if.

File: app/templates/components/settings/password-section.hbs [hunk 4]
```diff
@@ -3,21 +3,40 @@
...
+      {{input type='password' name='password_repeat' placeholder=(t 'Repeat Password') value=passwordRepeat}}
+    {{/if}}
   </div>
   <button class="ui teal button" type="submit">
     {{t 'Save'}}
   </button>
+  <button class="ui teal button" {{action 'showPassword'}}>
+    {{#if showPass}}
+      {{t 'Hide Password'}}
+    {{else}}
```
Comment (CosmicCoder96) [inline app/templates/components/settings/password-section.hbs]: use inline if.

**Comment by Anupam-dagar** [comment]
@CosmicCoder96 @pradeepgangwar please review.

**Comment by pradeepgangwar** [comment]
@Anupam-dagar Can you post new screenshots please? Will be quick to review then :)

**Comment by Anupam-dagar** [comment]
@pradeepgangwar My bad, here are the screenshots.
Screenshot from 2019-04-11 13-23-11
Screenshot from 2019-04-11 13-23-05

**Comment by Anupam-dagar** [comment]
@CosmicCoder96 I have implemented the required changes.
Screenshot from 2019-04-11 16-33-24

Screenshot from 2019-04-11 16-33-18

Screenshot from 2019-04-11 16-33-10

File: app/templates/components/settings/password-section.hbs [hunk 5]
```diff
@@ -3,21 +3,42 @@
...
+          <i class="{{if showPassRepeat 'hide' 'unhide'}} basic icon"></i>
+        </button>
+      </div>
+    </div>
   </div>
   <button class="ui teal button" type="submit">
     {{t 'Save'}}
   </button>
-</form>
+</form>
```
Comment (uds5501) [inline app/templates/components/settings/password-section.hbs]: Everything looks good, just add a new line at the end.  That should do

**Comment by Anupam-dagar** [comment]
@CosmicCoder96 Please review.

File: app/templates/components/settings/password-section.hbs [hunk 6]
```diff
@@ -3,19 +3,40 @@
...
   </div>
   <div class="field">
     <label>
       {{t 'New Password'}}
     </label>
-    {{input type='password' name='password_new' placeholder=(t 'New Password') value=passwordNew}}
+    <div class="ui icon input">
+      {{input type=(if showPassNew 'text' 'password') name='password_new' placeholder=(t 'New Password') value=passwordNew}}
+      <div class="ui small basic icon buttons">
+        <button type="button" class="ui button" id="eye1" {{action 'showPassword' 'New'}}>
```
Comment (CosmicCoder96) [inline app/templates/components/settings/password-section.hbs]: why is the `id` required?

File: app/templates/components/settings/password-section.hbs [hunk 7]
```diff
@@ -3,19 +3,40 @@
...
   </div>
   <div class="field">
     <label>
       {{t 'Repeat Password'}}
     </label>
-    {{input type='password' name='password_repeat' placeholder=(t 'Repeat Password') value=passwordRepeat}}
+    <div class="ui icon input">
+      {{input type=(if showPassRepeat 'text' 'password') name='password_repeat' placeholder=(t 'Repeat Password') value=passwordRepeat}}
+      <div class="ui small basic icon buttons">
+        <button type="button" class="ui button" id="eye1" {{action 'showPassword' 'Repeat'}}>
```
Comment (CosmicCoder96) [inline app/templates/components/settings/password-section.hbs]: two elements on the same page should never have the same id.

File: app/templates/components/settings/password-section.hbs [hunk 8]
```diff
@@ -3,19 +3,40 @@
     <label>
       {{t 'Current Password'}}
     </label>
-    {{input type='password' name='password_current' placeholder=(t 'Current Password') value=passwordCurrent}}
+    <div class="ui icon input">
+      {{input type=(if showPassCurrent 'text' 'password') name='password_current' placeholder=(t 'Current Password') value=passwordCurrent}}
+      <div class="ui small basic icon buttons">
+        <button type="button" class="ui button" id="eye1" {{action 'showPassword' 'Current'}}>
```
Comment (CosmicCoder96) [inline app/templates/components/settings/password-section.hbs]: Id's can't be the same.

**Comment by Anupam-dagar** [comment]
@CosmicCoder96 I have made the required changes. It was a reused piece of code and hence the `id` was there.

**Pull request merged**